tmpfs/tmpfs.go

```diff
@@ -0,0 +1,120 @@
+package tmpfs
+
+import (
+	"errors"
+	"os"
+	"path"
+
+	"gopkg.in/src-d/go-billy.v2"
+	"gopkg.in/src-d/go-billy.v2/subdirfs"
+)
+
+type tmpFs struct {
+	fs        billy.Filesystem
+	tmp       billy.Filesystem
+	tempFiles map[string]bool
```

Comment:
That would be could, although I think we need to check explicitely if it's temp file or not to provide the functionality of copying it to `fs` in some cases, so it's not only about resolving `Filsystem` that should be accessed.